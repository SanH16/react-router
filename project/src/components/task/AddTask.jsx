import { useState } from "react";
import { addTask } from "../../store/tasks";
import { useDispatch } from "react-redux";

let nextId = 3;

export default function AddTask() {
  const [text, setText] = useState("");
  // const dispatch = useDispatch();

  return (
    <>
      <input placeholder="Add task" value={text} onChange={(e) => setText(e.target.value)} />
      <button
        onClick={() => {
          // setText("");
          // dispatch(addTask({ id: nextId++, text }));
        }}
      >
        Add
      </button>
    </>
  );
}
