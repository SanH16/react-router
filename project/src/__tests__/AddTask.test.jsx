import { fireEvent, render, screen } from "@testing-library/react";
import AddTask from "../components/task/AddTask";

describe("AddTasks", () => {
  it("should render correctly without crash", () => {
    const { container } = render(<AddTask />);
    expect(container).toBeInTheDocument();
  });

  it("should match with snapshots", () => {
    const { container } = render(<AddTask />);
    expect(container).toMatchSnapshot();
  });

  it("input element must has placeholder", () => {
    render(<AddTask />);
    const inputElement = screen.getByPlaceholderText(/add task/i);
    expect(inputElement).toHaveAttribute("placeholder", "Add task");
  });

  it('button element must have text "Add"', () => {
    render(<AddTask />);
    const buttonElement = screen.getByText(/add/i);
    expect(buttonElement).toHaveTextContent("Add");
  });

  it("should trigger input element onchange correcty", () => {
    render(<AddTask />);
    const inputElement = screen.getByPlaceholderText(/add task/i);
    fireEvent.change(inputElement, { target: { value: "test" } });
    expect(inputElement).toHaveValue("test");
  });

  it("should trigger button element onclick correcty", () => {
    render(<AddTask />);
    const buttonElement = screen.getByText(/add/i);
    const inputElement = screen.getByPlaceholderText(/add task/i);

    fireEvent.click(buttonElement);
    expect(inputElement).toHaveValue("");
  });
});
