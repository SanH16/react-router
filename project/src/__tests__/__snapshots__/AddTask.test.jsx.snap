// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddTasks > should match with snapshots 1`] = `
<div>
  <input
    placeholder="Add task"
    value=""
  />
  <button>
    Add
  </button>
</div>
`;
